### DISCUSSION TITLE
Dual graph (How to center a vertex instead of a tile)

### DISCUSSION THREAD

**Discussion by Dib-Roy** [cduck/hyperbolic#11] [Q&A] [answered]
Thanks for the amazing package!

I have made the following graph,

![hyppent](https://user-images.githubusercontent.com/52301288/131942918-9857b3d1-71eb-4a89-8ad5-d9952e1e1dc2.png)

and am trying to draw its dual. I have so far been able to make (all this using the code given in the example files), 

![hypcay](https://user-images.githubusercontent.com/52301288/131942955-b5e3bc6c-9f27-4e4c-9a22-f7a1e1bec18d.png)
which is the dual but they do not line up as I want a vertex in the center. Is there an easy way to do this?

Thanks for the help!

**Comment by cduck** [comment, marked as answer]
Yes.  The tilings are generated by placing the center polygon then placing the rest adjacent to earlier tiles.  You can translate the first polygon off-center to get what you want.

It is easy to do this if you already use `layout.defaultStartTile` to generate the first tile.  See the example below (adapted from [tiles.ipynb](https://github.com/cduck/hyperbolic/blob/cd47fbc0bd804af5a4ccc1a2f57950d3c4e76624/examples/tiles.ipynb)).
```python
p1 = 4
p2 = 3
q = 3

theta1, theta2 = math.pi*2/p1, math.pi*2/p2
phiSum = math.pi*2/q
r1 = triangleSideForAngles(theta1/2, phiSum/2, theta2/2)
r2 = triangleSideForAngles(theta2/2, phiSum/2, theta1/2)

tGen1 = htiles.TileGen.makeRegular(p1, hr=r1, skip=1)
tGen2 = htiles.TileGen.makeRegular(p2, hr=r2, skip=1)

tLayout = htiles.TileLayout()
tLayout.addGenerator(tGen1, (1,)*p1)
tLayout.addGenerator(tGen2, (0,)*p2)
# centerCorner=True places the vertex at the center instead of the face
startTile = tLayout.defaultStartTile(rotateDeg=90, centerCorner=True)  # <- This line

tiles = tLayout.tilePlane(startTile, depth=4)

d = Drawing(2, 2, origin='center')
d.draw(euclid.shapes.Circle(0, 0, 1), fill='silver')
drawTiles(d, tiles)

d.setRenderSize(w=400)
d.saveSvg('example.svg')
d
```
![example output](https://user-images.githubusercontent.com/2476062/131945317-e8fdc137-c2ec-4f06-b461-706ce9c7b925.png)

If you need more customization, you can move the start tile:
```python
startTile = startTile.makeTransformed(Transform....())
```

**Reply by Dib-Roy** [reply]
Thank you so much!

Is there any documentation that I could refer to in the future?

**Reply by cduck** [reply]
Not right now.  Just tab completion in your editor and looking at the source code.  In Jupyter press tab while the cursor is at the end of a function name.
```python
startTile = tLayout.defaultStartTile(<press tab>
Transform.<press tab>
```
